app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -267,6 +272,33 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
         }
 
         selectedEmojiPack = preferences.getString(EMOJI_PREFERENCE, "")
+
+        onBackPressedDispatcher.addCallback(
+            this,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    when {
+                        binding.mainDrawerLayout.isOpen -> {
+                            binding.mainDrawerLayout.close()
+                        }
+                        binding.viewPager.currentItem != 0 -> {
+                            binding.viewPager.currentItem = 0
+                        }
+                        else -> {
+                            finish()
+                        }
+                    }
+                }
+            }
+        )
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(
```

Comment:
This logic very basic but I think it is enough? Android will not show the dialog if a user denied the Permission more than twice anyway.

Reply:
Seems fine